src/qibo/models/circuit.py

```diff
@@ -1423,3 +1423,71 @@ def chunkstring(string, length):
             output += table
 
         return output.rstrip("\n")
+
+
+class VariationalCircuit(Circuit):
```

Comment:
Maybe it should be better to define this in a separate file.

Reply:
Yes, let's see what @AleCandido and @scarrazza think

Reply:
In general, don't be afraid of using more files. It is fine to have even a file with 10 lines, if there is a good reason to keep it separate.

However, on the other side, it's not even convenient to have 1000 files with 10 lines each (even because most of them would be `import` statements).
So, your files should be approximately uniform: it's not a strict rule, they might be very small, or a bit larger than average. But never too large (as a rule of thumb, if it hits 1000 lines you might start considering which is the best way of splitting it).

In this specific case, this file is large because Python does not allow to define a class over multiple files (but other clean solutions are possible, such as splitting the most of the functions' bodies, and have some functions that operate on these objects, without being methods).
But since the addition of this PR is not part of the `Circuit` class, it's not a bad idea to put that one in a separate module.